slashing-protection/src/integration-test/java/tech/pegasys/web3signer/slashingprotection/WatermarkImportingIntegrationTest.java

```diff
@@ -0,0 +1,196 @@
+import tech.pegasys.web3signer.slashingprotection.interchange.model.Metadata;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.SignedAttestation;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import dsl.InterchangeV5Format;
+import dsl.SignedArtifacts;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.junit.jupiter.api.Test;
+
+public class WatermarkImportingIntegrationTest extends InterchangeBaseIntegrationTest {
+
+  final Bytes PUBLIC_KEY =
+      Bytes.fromHexString(
+          "0xb845089a1457f811bfc000588fbb4e713669be8ce060ea6be3c6ece09afc3794106c91ca73acda5e5457122d58723bed");
+  final int VALIDATOR_ID = 1;
+
+  @Test
+  public void importSetsBlockWatermarkToLowestInImportWhenDatabaseIsEmpty()
+      throws JsonProcessingException {
+    final InputStream input = createInputDataWith(List.of(5, 4), emptyList());
+    slashingProtection.importData(input);
+    assertThat(getWatermark(VALIDATOR_ID))
+        .isEqualToComparingFieldByField(
+            new SigningWatermark(VALIDATOR_ID, UInt64.valueOf(4), null, null));
+  }
+
+  @Test
+  public void
+      blockWaterMarkIsUpdatedIfImportHasSmallestMinimumBlockLargerValueThanExistingMaxBlock()
+          throws JsonProcessingException {
+    final UInt64 initialBlockSlot = UInt64.valueOf(3);
+    insertValidator(PUBLIC_KEY, VALIDATOR_ID);
+    slashingProtection.registerValidators(List.of(PUBLIC_KEY));
+    insertBlockAt(initialBlockSlot, PUBLIC_KEY);
+    insertBlockAt(UInt64.valueOf(10), PUBLIC_KEY); // max = 10, watermark = 3 (As was first).
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(initialBlockSlot);
+
+    final InputStream input = createInputDataWith(List.of(20, 19), emptyList());
+    slashingProtection.importData(input);
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(UInt64.valueOf(19));
+  }
+
+  @Test
+  public void blockWatermarkIsNotUpdatedIfLowestBlockInImportIsLowerThanHighestBlock()
+      throws JsonProcessingException {
+    insertValidator(PUBLIC_KEY, VALIDATOR_ID);
+    slashingProtection.registerValidators(List.of(PUBLIC_KEY));
+    insertBlockAt(UInt64.valueOf(3), PUBLIC_KEY);
+    insertBlockAt(UInt64.valueOf(10), PUBLIC_KEY);
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(UInt64.valueOf(3));
+
+    final InputStream input = createInputDataWith(List.of(6, 50), emptyList());
+    slashingProtection.importData(input);
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(UInt64.valueOf(3));
+  }
+
+  @Test
+  public void watermarkDoesNotMoveIfAnyImportedBlockIsLessThanPriorMax()
```

Comment:
It would be safe to move the watermark in this case, but we have decided not to - i.e. the import data is explicitly implying that it did not sign intermediate entires - so we're safe to do so.